src/Storages/StorageMerge.h

```diff
@@ -159,6 +159,8 @@ class ReadFromMerge final : public SourceStepWithFilter
     StoragePtr storage_merge;
     StorageSnapshotPtr merge_storage_snapshot;
 
+    std::vector<QueryPlan> child_plans;
```

Comment:
No, it's not a good idea.
Storage may leave for a long time, but we don't clean these plans.